static/app/views/organizationIntegrations/sentryAppExternalForm.tsx

```diff
@@ -135,6 +146,28 @@ export class SentryAppExternalForm extends Component<Props, State> {
     }
   };
 
+  getDefaultOptions = (field: FieldFromSchema) => {
+    const savedOption = ((this.props.resetValues || {}).settings || []).find(
+      value => value.name === field.name
+    );
+    const currentOptions = (field.choices || []).map(([value, label]) => ({
+      value,
+      label,
+    }));
+
+    const shouldAddSavedOption =
+      // We only render saved options if they have preserved the label, otherwise it appears unselcted.
+      // The next time the user saves, the label should be preserved.
```

Comment:
What I mean by this is even if we have a saved value, if there isn't a label, it should render blank. This will be the case for existing alert rules prior to landing this PR.

Once this is in place, any edits to the rule will persist the label. This allows this change to be backwards compatible.